src/zenml/integrations/kubernetes/deployers/kubernetes_deployer.py

```diff
@@ -0,0 +1,1009 @@
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize the Kubernetes deployer.
+
+        Args:
+            **kwargs: Additional keyword arguments.
+        """
+        super().__init__(**kwargs)
+
+        self._ctx: Optional[_DeploymentCtx] = None
+        self._k8s_client: Optional[k8s_client.ApiClient] = None
+        self._engine: Optional[KubernetesTemplateEngine] = None
+        self._applier: Optional[KubernetesApplier] = None
+
+    def require_ctx(self) -> _DeploymentCtx:
+        """Require the deployment context.
+
+        Returns:
+            The deployment context.
+
+        Raises:
+            DeployerError: If the deployment context is not initialized.
+        """
+        if self._ctx is None:
+            raise DeployerError("Deployment context not initialized.")
+        return self._ctx
+
+    # ========================================================================
+    # Kubernetes Client Management
+    # ========================================================================
+
+    def get_kube_client(
+        self, incluster: Optional[bool] = None
+    ) -> k8s_client.ApiClient:
+        """Get authenticated Kubernetes client.
+
+        Args:
+            incluster: Whether to use in-cluster config.
+
+        Returns:
+            Authenticated Kubernetes API client.
+
+        Raises:
+            RuntimeError: If connector returns invalid client type.
+        """
+        if incluster is None:
+            incluster = self.config.incluster
+
+        if incluster:
+            kube_utils.load_kube_config(
+                incluster=incluster,
+                context=self.config.kubernetes_context,
+            )
+            self._k8s_client = k8s_client.ApiClient()
+            return self._k8s_client
+
+        if self._k8s_client and not self.connector_has_expired():
+            return self._k8s_client
+
+        connector = self.get_connector()
+        if connector:
+            client = connector.connect()
+            if not isinstance(client, k8s_client.ApiClient):
+                raise RuntimeError(
+                    f"Expected k8s_client.ApiClient but got {type(client)}"
+                )
+            self._k8s_client = client
+        else:
+            kube_utils.load_kube_config(
+                incluster=incluster,
+                context=self.config.kubernetes_context,
+            )
+            self._k8s_client = k8s_client.ApiClient()
+
+        return self._k8s_client
+
+    @property
+    def k8s_core_api(self) -> k8s_client.CoreV1Api:
+        """Get Kubernetes Core V1 API client.
+
+        Returns:
+            Kubernetes Core V1 API client.
+        """
+        return k8s_client.CoreV1Api(self.get_kube_client())
+
+    @property
+    def k8s_applier(self) -> KubernetesApplier:
+        """Get or create Kubernetes Applier instance.
+
+        Returns:
+            Kubernetes Applier instance.
+        """
+        if not self._applier:
```

Comment:
FYI this will ignore service connector expiration. It should probably include this at the top:

```
if self.connector_has_expired():
    self._applier = None
```